### PR TITLE
[xorg] virtual package for xorg (X11) libraries

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **xorg/system**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

continues the story of #1535 
~~(currently blocked by https://github.com/conan-io/hooks/issues/193)~~
required for [glwf](https://github.com/conan-io/conan-center-index/pull/1242) and others

### CODE REVIEW

File: recipes/xorg/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class ConanXOrg(ConanFile):
+    name = "xorg"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    homepage = "https://www.x.org/wiki/"
+    description = "The X.Org project provides an open source implementation of the X Window System."
+    settings = {"os": "Linux"}
+
+    def package_id(self):
```
Comment (uilianries) [inline recipes/xorg/all/conanfile.py:13]: I'm curious why header_only, if we can use only one configuration (os=Linux)? Is it necessary?

File: recipes/xorg/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    def _libs_pkgconfig(self, name):
+        pkg_config = tools.PkgConfig(name)
+        if not pkg_config:
+            raise ConanException("%s development files aren't available, give up" % name)
+        libs = [lib[2:] for lib in pkg_config.libs_only_l]
+        return libs
+
+    def system_requirements(self):
+        if tools.os_info.is_linux and self.settings.os == "Linux":
```
Comment (uilianries) [inline recipes/xorg/all/conanfile.py:24]:
```suggestion
        if tools.os_info.is_linux:
```

line 11 only allows Linux

File: recipes/xorg/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            if tools.os_info.with_apt:
+                packages = ["xorg-dev"]
+            elif tools.os_info.with_yum:
+                packages = ["xorg-x11-server-devel"]
+            elif tools.os_info.with_pacman:
+                packages = ["xorg-server-devel"]
+            elif tools.os_info.with_zypper:
+                packages = ["Xorg-x11-devel"]
+            else:
+                self.warn("don't know how to install xorg for your distro")
```
Comment (uilianries) [inline recipes/xorg/all/conanfile.py:35]:
```suggestion
                self.warn("Do not know how to install 'xorg' for {}.".format(tools.os_info.linux_distro))
```

File: recipes/xorg/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/xorg/all/conanfile.py:13]: to erase the `os` setting

File: recipes/xorg/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/xorg/all/conanfile.py:24]: I'd like to keep it, if we ever add support for FreeBSD/NetBSD/OpenBSD, which will use the same logic

File: recipes/xorg/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (jgsogo) [inline recipes/xorg/all/test_package/conanfile.py:7]: Does it work with the `cmake_find_package`... As it has no proper `libs` I would like to check the targets are created.

File: recipes/xorg/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/xorg/all/conanfile.py:24]: (also macOS which may have X11 installed via Xquartz)
and in both cases, the same package (which is literally a package containing zero files) should be available for Linux/*BSD/macOS/whatever

File: recipes/xorg/all/test_package/conanfile.py (hunk 4 above)
Comment (SSE4) [reply recipes/xorg/all/test_package/conanfile.py:7]: > What are the outputs of this recipe when the libraries are not installed in the system? What is the feedback to the user?
```
The following packages need to be installed:
xorg-dev
Aborted due to CONAN_SYSREQUIRES_MODE=verify
Some system packages need to be installed
```

**Comment by SSE4** [comment]
> What are the outputs of this recipe when the libraries are not installed in the system? What is the feedback to the user?
```
The following packages need to be installed:
xorg-dev
Aborted due to CONAN_SYSREQUIRES_MODE=verify
Some system packages need to be installed
```

**Comment by uilianries** [comment]
@SSE4 Is it still a draft and blocked?

**Comment by uilianries** [comment]
@SSE4 the system_libs output repeats X11 many times:

```
set(CONAN_SYSTEM_LIBS X11 dmx fontenc FS ICE SM Xau Xaw7 Xt X11 Xcomposite Xcursor Xdamage Xfixes Xdmcp Xext Xfixes Xft Xi Xinerama xkbfile Xmu Xt X11 Xmuu Xpm X11 Xrandr Xrender X11 XRes Xss Xt X11 Xtst Xv XvMC Xv Xxf86dga Xxf86vm ${CON
```

Is it correct?

**Comment by SSE4** [comment]
@uilianries I think so, it's because of the transitive dependencies (e.g. X11 is dependency of several other top-level libraries)

**Comment by SSE4** [comment]
@ericLemanissier if there is such risk, it already exists for consumers who are running on Linux systems with that packages installed. if there are such situations, we have to fix them either way.

**Comment by ericLemanissier** [comment]
I am thinking of two specific issues, which don't happen if the libraries just happen to be installed on system: if a package depends on xorg/system, one of the xorg libs will try to load system fontconfig (for example). what happens if the file name of the system fontconfig library is not the same as conan fontconfig package ? Also, won't the system fontconfig be always chosen, because it is present in the executable's rpath ?

**Comment by SSE4** [comment]
@jgsogo @czoido need your review, then we can finally add `glfw` and `glew`

**Pull request merged**